webapp/src/main/java/com/box/l10n/mojito/service/security/user/UserRepository.java

```diff
@@ -37,4 +39,21 @@ public interface UserRepository extends JpaRepository<User, Long>, JpaSpecificat
   @Override
   @EntityGraph(value = "User.legacy", type = EntityGraphType.FETCH)
   Optional<User> findById(Long aLong);
+
+  @Query(
+      """
+      select u
+      from User u
+      where (:username is null or u.username = :username)
+        and ((:search is null or :search = '')
+             or (lower(u.username) like lower(concat('%', :search, '%'))
+                 or (u.commonName is not null
+                     and u.commonName <> ''
+                     and lower(u.commonName) like lower(concat('%', :search, '%')))
+                 or ((u.commonName is null or u.commonName = '')
+                      and lower(concat(u.givenName, ' ', u.surname)) like lower(concat('%', :search, '%')))))
+        and u.enabled = true
+      """)
+  Page<User> findAll(
```

Comment:
Is `findAll` appropriately named, query looks be be retrieving all users or by username/name if that's optionally provided?

Reply:
Renamed to findByUsernameOrName, please review again